generator/src/main/java/com/scottlogic/deg/generator/restrictions/linear/LinearRestrictionsMerger.java

```diff
@@ -0,0 +1,105 @@
+package com.scottlogic.deg.generator.restrictions.linear;
+
+import com.scottlogic.deg.generator.restrictions.MergeResult;
+
+public class LinearRestrictionsMerger<T extends Comparable<T>> {
+
+    public MergeResult<LinearRestrictions<T>> merge(LinearRestrictions<T> left, LinearRestrictions<T> right){
+        if (left == null && right == null)
+            return new MergeResult<>(null);
+        if (left == null)
+            return new MergeResult<>(right);
+        if (right == null)
+            return new MergeResult<>(left);
+
+        Granularity<T> mergedGranularity = left.getGranularity().merge(right.getGranularity());
+
+        Limit<T> mergedMin = getHighest(left.getMin(), right.getMin(), mergedGranularity);
+        Limit<T> mergedMax = getLowest(left.getMax(), right.getMax(), mergedGranularity);
+
+        LinearRestrictions<T> mergedRestriction = new LinearRestrictions<>(mergedMin, mergedMax, mergedGranularity, left.getConverter());
+
+        if (isContradictory(mergedRestriction)) {
+            return MergeResult.unsuccessful();
+        }
+
+        return new MergeResult<>(mergedRestriction);
+    }
+
+    private boolean isContradictory(LinearRestrictions<T> restictions) {
+        Limit<T> min = restictions.getMin();
+        Limit<T> max = restictions.getMax();
+
+        if (min == null || max == null){
+            return false;
+        }
+
+        if (min.getValue().equals(max.getValue())){
+            if (!min.isInclusive()){
+                return true;
+            }
+            if (!max.isInclusive()){
+                return true;
+            }
+        }
+
+        return !min.isBefore(max.getValue());
+    }
+
+    private Limit<T> getHighest(Limit<T> left, Limit<T> right, Granularity<T> granularity) { //TODO dry this code up
```

Comment:
Deal with this TODO
Pull it out into a function that returns a `Optional` so this function becomes:
```
return NewFunction(left, right).orElse(
     roundUpToNextValidValue(
        left.isBefore(right.getValue()) ? left : right,
        granularity
    ));
```